### ISSUE TITLE
F11 map and blips replace via script

### ISSUE THREAD

**Issue by etrnl1337** [multitheftauto/mtasa-blue#2574]
There is no way to replace F11 map parts or the whole map and blips. There is an option to replace ( use Shaders ) to apply new graphics to HUD ( for weapons, fist etc ) and to the bottom left radar to replace map parts and blips textures.

I know everyone can toggle off F11 and use DX functions to draw their custom map, but i want to suggest that f11 blips and map replacements should be added to optimize a lot of servers and to make them more configurable.

It's very frustrating that i need to unbind f11 and rewrite from scratch this f11, and also f11 map have much options like zoom in/out, toggle opacity, focusing on localPlayer etc that are very hard to implement.

Please consider adding this in next MTA update as it would make a lot of servers much more optimized and good looking.

**Comment by Vadya963** [comment]
https://community.multitheftauto.com/index.php?p=resources&s=details&id=381
complete replacement of the card with all the features

**Comment by etrnl1337** [comment]
> https://community.multitheftauto.com/index.php?p=resources&s=details&id=381 complete replacement of the card with all the features

It's okay for beginners and small servers. It lacks optimization. 

Getting radar areas / blips / other players 30/60 times a second (because of onClientPreRender) is such a silly method that i won't comment it. Using element data for blips that should not draw (?), rendering images instead of textures. It's okay for someone who don't draw attention to details and client optimization. I'am able to write better script but that's not the point - every script won't ever be better than hardcoded F11.

**Comment by PlatinMTA** [comment]
> > https://community.multitheftauto.com/index.php?p=resources&s=details&id=381 complete replacement of the card with all the features
> 
> It's okay for beginners and small servers. It lacks optimization.

It's an script from 2009, chill. It's a really good base that you can improve up on (and you are blowing out of proportion it's optimization, you can make it run better in 15 minutes without problem).

If your concern are the new scripters then you should wait until someone wants to work on this or you could work on this yourself, but I don't know if something related to this will get merged.

A thing that should be worked on is the ability to disable the F11 map, because disabling the ``radar`` control will sometimes not work or it will work but the client already had it open. ``forcePlayerMap`` only works for forcing it on, it can't close one that it's already open.

**Comment by etrnl1337** [comment]
> > > https://community.multitheftauto.com/index.php?p=resources&s=details&id=381 complete replacement of the card with all the features
> > 
> > 
> > It's okay for beginners and small servers. It lacks optimization.
> 
> It's an script from 2009, chill. It's a really good base that you can improve up on (and you are blowing out of proportion it's optimization, you can make it run better in 15 minutes without problem).
> 
> If your concern are the new scripters then you should wait until someone wants to work on this or you could work on this yourself, but I don't know if something related to this will get merged.
> 
> A thing that should be worked on is the ability to disable the F11 map, because disabling the `radar` control will sometimes not work or it will work but the client already had it open. `forcePlayerMap` only works for forcing it on, it can't close one that it's already open.

Yup, for a 2009 script it's very good. As i've told of course everyone can improve it but it never will work as hardcoded F11. That's strange that left down radar and HUD can be improved by shaders and textures and F11 not.  Problem here is that me and also many other scripters need to spend a lot of time to make it work perfectly where their goal is let's say replacing one blip icon. If i knew how to implement this to MTA i would do it a long time ago. Thanks for your comment and also thanks for sharing with this toggleControl bug (i didn't know about this one).

**Comment by Proxy-99** [comment]
may related to #2554
